hoot-core/src/main/cpp/hoot/core/conflate/splitter/DualWaySplitter.cpp

```diff
@@ -117,8 +118,10 @@ boost::shared_ptr<Way> DualWaySplitter::_createOneWay(boost::shared_ptr<const Wa
   boost::shared_ptr<Geometry> g(bb.bufferLineSingleSided(ls.get(), bufferSize, left));
   const LineString* newLs = dynamic_cast<const LineString*>(g.get());
 
-  boost::shared_ptr<Way> result(new Way(w->getStatus(), _result->createNextWayId(),
-    w->getRawCircularError()));
+  long way_id = w->getId();
+  if (!left)
+    way_id = _result->createNextWayId();
+  boost::shared_ptr<Way> result(new Way(w->getStatus(), way_id, w->getRawCircularError()));
```

Comment:
Here we keep the left hand side of the road the original way id and create a new one for the right hand side.